enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,260 @@
+---
+title: bundle-upgrade
+authors:
+  - "@rashmigottipati"
+reviewers:
+  - "@joelanford"
+  - "@estroz"
+  - "@shawn-hurley"
+  - "@jmrodri"
+approvers:
+  - TBD
+creation-date: 2020-06-11
+last-updated: 2020-06-11
+
+---
+
+# Bundle Upgrade
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [X] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions
+
+1. How do we handle multiple subscriptions? Are subscriptions patch-able ex. can the starting CSV or channel be updated for the same subscription?
+    * One of the assumptions of this proposal is that a previous version of an operator is not running so we would be creating a subscription from scratch using the starting CSV and channel provided on the command line. Assuming happy path, that would cause OLM to install the operator defined by the starting CSV and then proceed through the upgrade graph for the defined channel. Assuming the provided bundle was inserted into the graph at the head of the channel, the OLM upgrades would stop after upgrading to the provided bundle. So we won't run into a scenario with an existing subscription and the need to patch it. 
+2. Should we explicitly pass new version number as a flag or can we get it from bundle image?
+    * Based on discussions, can extract the version number from bundle image. 
+
+## Summary
+
+Enhance the Operator SDK to add a new subcommand `run bundle-upgrade` that performs an operator upgrade using OLM.
+
+## Motivation
+
+Testing operator upgrades remains difficult today due to the number of steps to prepare for an upgrade and the lack of tooling that automates those steps. Having high quality Operators is crucial to the success of the Operator Framework and thereby operator developers should have a tool to use to perform an operator upgrade.
+
+### Goals
+
+Provide a command line that upgrades an operator version using OLM.
+
+### Non-Goals
+
+This enhancement doesn't include the capability for SDK to use labels to detect the DB path. For the sake of simplicity, we will assume that the existing registry DB is located at `/database/index.db`.
+
+Another non-goal is that the enhancement is intended for operators that don't exist in the cluster. The scenario in which existing operators exist in the cluster is out of scope for this proposal.
```

Comment:
If this feature is limited like this, then we can not test that operands will be upgraded by the new version of an operator.

Reply:
Thanks for your input Matskiv. Updated the PR with revised proposal which can upgrade from a previously installed operator to a new specified version. Please take a look.

Reply:
Thank you for addressing use case for my team.
I skimmed through the proposal and it looks good to me, but your team is, of course, in a better position to properly asses technical side of the proposal. :)

Reply:
thank you so much for your valuable feedback and explaining your use case in our offline discussion. glad that this will feature will be useful for your use case!